tests/test_metric.py

```diff
@@ -46,3 +49,87 @@ def test_sanitize_metric_name_should_handle_empty_value(self):
         self.assertEqual("", bg_metric.sanitize_metric_name(""))
         self.assertEqual("", bg_metric.sanitize_metric_name("."))
         self.assertEqual("", bg_metric.sanitize_metric_name(".."))
+
+
+class TestMetricMetadata(unittest.TestCase):
```

Comment:
I feel like naming tests describing the expected behavior provides more info about the test, with a "thing SHOULD do stuff WHEN something". Eg:

* `test_create_should_define_default_values`
* `test_metadata_should_be_unique_for_a_given_parameter_tuple`
* `test_xfilesfactor_should_have_no_impact_on_metadata_uniqueness`
* ...

What is your thought about it?